maven-surefire-common/src/main/java/org/apache/maven/plugin/surefire/AbstractSurefireMojo.java

```diff
@@ -2569,7 +2569,7 @@ private TestClassPath generateTestClasspath() throws MojoFailureException {
 
         Map<String, Artifact> dependencyConflictIdsProjectArtifacts = classpathArtifacts.stream()
                 .collect(Collectors.toMap(Artifact::getDependencyConflictId, Function.identity()));
-        Set<String> additionalClasspathElements = new HashSet<>();
+        Set<String> additionalClasspathElements = new LinkedHashSet<>();
```

Comment:
LGTM, but a test case would be really appreciated so that we prevent regressions in the future.

Reply:
I can find some time to do this next week hopefully.